### PR TITLE
Fix dynamic connector extensions & update internal testing

### PR DESCRIPTION
**Pull request by LucGenetier**
Fix dynamic connector extensions management in making 'parameters' optional despite the spec which says it's required
Update internal tests
Add Guardian files
Update .gitignore file
Fix BuildLocalPackages.cmd script

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: .gdn/.gdnsettings [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (MikeStall) [inline .gdn/.gdnsettings:1]: what is this?

File: src/buildLocalPackages.cmd [hunk 2]
```diff
@@ -2,11 +2,10 @@
 setlocal enabledelayedexpansion
 
 set MSBUILDARGS=-p:PublishRepositoryUrl=true -p:GeneratePackages=true -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg -p:InternalBuild=true -p:Configuration=Debug -p:Platform="Any CPU"
+set MSBUILD=C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\MSBuild.exe
 
-for /f "usebackq tokens=*" %%i in (`"%programfiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do (
```
Comment (MikeStall) [inline src/buildLocalPackages.cmd:6]: What is this? (did you intend to check in)

Comment (LucGenetier) [reply src/buildLocalPackages.cmd:6]: Yes, this script doesn't work anymore so I fixed it
It used to search for latest VS version but this approach gives you VS2019 and not VS2022

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**